SourceCode/raytracer/Renderer.h

```diff
@@ -12,12 +12,92 @@ class Renderer {
 
     void render(Buffer2D& image) const
     {
-
+        for (int y = 0; y < camera.getHeight(); ++y) {
+            for (int x = 0; x < camera.getWidth(); ++x) {
+                Ray ray = camera.generateRay(x, y);
+                // Color color = traceImagePlane(ray); // hw3
+                Color color = traceRay(ray); // hw5
+                image(x, y) = color;
+            }
+        }
     }
 
 private:
     const Camera& camera;
     // const Scene& scene; // TODO: Implement Scene class
 
+    Color traceRay(const Ray& ray) const
+    {
+        
+        // alls vertices even with the image plane
+        // This one does not cause problems
+        Triangle tri_even
+        {
+            Vec3{-1.f, -1.f, -1.f},
+            Vec3{1.f, -1.f, -1.f},
+            Vec3{0.f, 1.f, -1.f},
+        };
+
+        // Bottom vert deeper
+        Triangle tri_bot
+        {
+            Vec3{-1.f, -1.f, -1.f},
+            Vec3{1.f, -1.f, -1.f},
+            Vec3{0.f, 1.f, -1.5f}
```

Comment:
Also, you say that the bottom vert is supposed to be deeper here, but in fact you have moved the vertex with the highest `y` component deeper. In everything else you have implemented, you consider `Y (0, 1, 0)` the up direction. So this is actually a triangle pointing upwards, with the top vertex pushed deeper (further from the origin).

You probably got confused because of the other mistake with the negative ray distance, flipping your image upside down.